client/editorQueryEvaluator.ts

```diff
@@ -0,0 +1,221 @@
+'use strict';
+import { ExtensionContext, TextDocument, TextEdit, TextEditor, Uri, WorkspaceEdit,
+    commands, window, workspace, WorkspaceConfiguration } from 'vscode';
+import { cascadeOverrideClient } from './vscQueryParameterTools';
+import { ClientContext, sendJSQuery, sendSparql, sendXQuery, sendRows } from './marklogicClient';
+import { ClientResponseProvider } from './clientResponseProvider';
+import { contentType, RowsResponse } from 'marklogic';
+import * as ml from 'marklogic';
+
+export enum EditorQueryType {
+    JS,
+    XQY,
+    SPARQL,
+    SQL,
+    ROWS
+}
+
+export class EditorQueryEvaluator {
+    private static FOPTIONS = { tabSize: 2, insertSpaces: true };
+    private static FCOMMAND = 'vscode.executeFormatDocumentProvider';
+    private static SJS = 'sjs';
+    private static XQY = 'xqy';
+
+    private context: ExtensionContext;
```

Comment:
I think `extensionContext` would be better now that we have two contexts flying around.